### PR TITLE
Update bescort.lic

### PR DESCRIPTION
**Pull request by bastionwt**
For people < 400 athletics, the ferry from Crossing to Leth can be a major time sink where little training is done. This update adds in first-aid to that downtime if a textbook or compendium is present, otherwise it defaults back to performance. No new settings are needed in the YAML, as both compendium_type and textbook_type are already present in base.yaml.

### CODE REVIEW

**Comment by bastionwt** [comment]
The build failing, I can't see details on what might need to be changed to allow that to succeed.

**Comment by rpherbig** [comment]
A timeout failed. I'm rerunning the build now.

File: bescort.lic [hunk 1]
```diff
@@ -1447,6 +1447,8 @@ class Bescort
   end
 
   def take_xing_ferry(mode)
+    compendium_type = get_settings.compendium_type
```
Comment (rpherbig) [inline bescort.lic:1450]: I don't think we want to get calling `get_settings` this many times. Could you cache it?

File: bescort.lic [hunk 2]
```diff
@@ -1462,9 +1464,22 @@ class Bescort
 
     case bput('go ferry', 'The Captain gives you a little nod', 'You hand him', 'The ferry has just pulled away from the dock', 'There is no ferry here to go aboard', 'Come back when you can afford the fare')
     when 'You hand him', 'The Captain gives you a little nod'
-      hide?
-      waitfor 'reaches the dock and its crew ties the ferry off'
-      move 'go dock'
+      # hide?
```
Comment (rpherbig) [inline bescort.lic:1467]: Why was this removed?

File: bescort.lic [hunk 3]
```diff
@@ -1462,9 +1464,22 @@ class Bescort
...
+      if get_settings.textbook
+        case bput("tap my #{textbook_type}", 'You tap', 'I could not', 'What were')
```
Comment (rpherbig) [inline bescort.lic:1469]: Could `DRCI.exists?` simplify this code a bit?

File: bescort.lic [hunk 4]
```diff
@@ -1474,6 +1489,30 @@ class Bescort
     end
   end
 
+  def ferry_textbook
+    start_script('first-aid')
```
Comment (rpherbig) [inline bescort.lic:1493]: I'm a little worried about calling `first-aid` because it may try to move around. For example: https://github.com/rpherbig/dr-scripts/blob/master/first-aid.lic#L19

What do you think?

File: bescort.lic [hunk 5]
```diff
@@ -1474,6 +1489,30 @@ class Bescort
...
+    waitfor 'reaches the dock and its crew ties the ferry off'
+    stop_script('first-aid') if Script.running?('first-aid')
+    waitrt?
+    move 'go dock'
+  end
+
+  def ferry_compendium
```
Comment (rpherbig) [inline bescort.lic:1500]: What do you think about using a single function that takes the script name as an argument?

**Comment by vtcifer** [comment]
General comment:

I feel like this should be an option, and not assumed one always wants to be out of hiding on the ferry.  It essentially overrides `bescort_hide: true`, unless I missed something.

File: bescort.lic (hunk 1 above)
Comment (bastionwt) [reply bescort.lic:1450]: I tried a couple different methods that didn't work. Do you have suggestions on how to implement?

File: bescort.lic (hunk 4 above)
Comment (bastionwt) [reply bescort.lic:1493]: I was unaware of the bleed bot setting. That's not something I've ever had enabled on this side. Suggestions?

File: bescort.lic (hunk 2 above)
Comment (bastionwt) [reply bescort.lic:1467]: Adding this back in as an option if they have bescort_hide: true

**Comment by bastionwt** [comment]
> General comment:
> 
> I feel like this should be an option, and not assumed one always wants to be out of hiding on the ferry. It essentially overrides `bescort_hide: true`, unless I missed something.

Yep. Updated to that affect, honoring the bescort option where present.

File: bescort.lic (hunk 4 above)
Comment (rpherbig) [reply bescort.lic:1495]: A few options I can think of offhand:
1. Return if `settings.bleed_bot`
2.  Add a 'nomove' mode to `first-aid` (optional script argument, I would think)
3. Add a 'no-bleedbot' mode to `first-aid` (similar but slightly different intent to option 2)

File: bescort.lic (hunk 1 above)
Comment (rpherbig) [reply bescort.lic:1450]: The simplest would be something like

```
settings = get_settings
compendium_type = settings.compendium_type
textbook_type = settings.textbook_type
use_textbook = settings.textbook
```
